app/models/validations/household_validations.rb

```diff
@@ -121,6 +121,14 @@ def validate_layear(record)
     end
   end
 
+  def validate_housing_needs(record)
+    if record.housingneeds == 1 && record.housingneeds_type == 3 && record.housingneeds_other&.zero?
+      record.errors.add :housingneeds, I18n.t("validations.household.housingneeds.invalid")
+      record.errors.add :housingneeds_type, I18n.t("validations.household.housingneeds.invalid")
+      record.errors.add :housingneeds_other, I18n.t("validations.household.housingneeds.invalid")
+    end
+  end
```

Comment:
my preference also in these validations is to have methods on the log class that make these codes human readable.
I don't know what these codes represent without going and checking elsewhere